accepted/0000-caasp-integration.md

```diff
@@ -0,0 +1,102 @@
+Uyuni/SUSE Manager must check if the underlying container engine is running Docker (and use the existing Kubernetes integration) or CRI-O (and adapt the current implementation to support CRI-O).
+The Kubernetes implementation that this RFC is targeting is the one in CaaS Platform version 4: 1.15.2. Other Kubernetes distributions will not be targeted, at least initially.
+
+Into the Virtual Host Manager, when a Kubernetes cluster is selected, the properties and the names of the nodes are shown, together with CPU Arch, CPU Sockets and RAM of the node.
+
+After the user has imported the `kubeconfig`, obtained from CaaS Platform, the process would be:
+
+- When a user registers a new cluster, we introduce a new workflow to bootstrap (using Systems > Bootstrapping) each node of the cluster
+- When every cluster node is registered, a system group with the same name of the cluster is created and all registered systems are moved into this system group
+- Each node presented the Nodes information is a link to the system overview page of that node
+
+## Documentation
+
+In addition to all the above, we are enriching our documentation on the CaaS Platform deployment. In the following sections, we are going to describe what we should introduce in the documentation.
+
+### AutoYaST Deployment documentation
+
+CaaS Platform provides AutoYaST templates for deploying a standard cluster (in the `SUSE-CaaSP-Management` package) comprising of:
+
+- 2 Load Balancers
+- 3 Masters
+- 3 Workers
+
+In the Uyuni/SUSE Manager documentation, we need to provide a step-by-step guide on how to customize the AutoYaST template, import it in Uyuni/SUSE Manager and deploy it to bare-metal machines.
+
+## Next steps
+
+Possible next steps are presented as additional "stretch goals": if time permits, those goals can be achieved in the scope of the initial integration.
+
+### Minion blackout configuration
+
+If the [minion blackout](https://docs.saltstack.com/en/latest/topics/blackout/) is feasible, we can set to blackout every minion registered to Uyuni/SUSE Manager that is part of a Kubernetes or CaaS Platform cluster. Every minion that is in blackout mode will reject all incoming commands.
+
+### Container registry documentation
+
+In the documentation, we can write a guide on how to set up a container registry (using a standard container registry or Portus) and mirror SUSE container registry.
+
+### Management node
+
+As part of the support for the CaaS Platform version 4, SUSE Manager can elect (via the Add-On System Type) a node as "Management Node" for the cluster. This node must run SUSE Linux Enterprise Server and have `skuba` (a tool packaged by CaaS Platform) to deploy and manage the node.
+Alternatively, we can choose to run `skuba` on the SUSE Manager server, we should add it to the packaging pattern of SUSE Manager server. The package is Go-vendored, no other dependency is required.
```

Comment:
Please check back with consultants if shell access to SUSE Manager for k8s cluster admins is feasible.

Reply:
@paususe do you have any champion consultant to suggest here?

Reply:
@kkaempf 

I'm not sure I understand the requirement. 

Do you mean we should create a local account in the SUSE Manager Server for each k8s cluster admin? With what rights (e. g. groups)? Should they also have shell access to the Proxies?

Reply:
If you have `skuba` on a Uyuni server, the caasp admin needs shell access to the Uyuni server in order to access and run `skuba`.
However, shell access to the Uyuni server could also be used for other (malicious) things beyond cluster management. And it breaks separation of concerns.

Reply:
Yes, I agree: the Management Node must be the server with `skuba` instead of the Uyuni/SUSE Manager one.

Reply:
Added the above to the alternative section, with the reason of why we don't want `skuba` to run on the SUSE Manager server.